tests/es6-class-name-lex-open-heritage.js

```diff
@@ -0,0 +1,57 @@
+// -*- coding: utf-8 -*-
+//  Copyright (C) 2014 Yusuke Suzuki <utatane.tea@gmail.com>
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are met:
+//
+//    * Redistributions of source code must retain the above copyright
+//      notice, this list of conditions and the following disclaimer.
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+//  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+//  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+//  ARE DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+//  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+//  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+//  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+//  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+//  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+//  THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+/* eslint-disable no-unused-expressions */
+
+import { expect } from "chai";
+import espree from "./util/espree.js";
+import { analyze } from "../lib/index.js";
+
+describe("ES6 class", () => {
+    it("declaration name creates class scope", () => {
```

Comment:
I think the new test(s) can be just added to [es6-class.js](https://github.com/eslint/eslint-scope/blob/main/tests/es6-class.js)

Reply:
There are two reasons why I did not do this: 
first, I want to ensure that this pull request passes normally even for existing test cases. 
second, is that if you want to change the existing test-case, you need to adjust the logic in more assertions, which is a bit heavy.